revocation/reasons.go

```diff
@@ -44,3 +50,32 @@ var UserAllowedReasons = map[Reason]struct{}{
 	Superseded:           {}, // superseded
 	CessationOfOperation: {}, // cessationOfOperation
 }
+
+// UserAllowedReasonsMessage creates a string describing a list of user allowed
+// revocation reasons. This is useful when a revocation is rejected because it
+// is not a valid user supplied reason and the allowed values must be
+// communicated.
+func UserAllowedReasonsMessage() string {
+	// Build a slice of ints from the allowed reason codes.
+	// We want a slice because iterating `UserAllowedReasons` will change order
+	// and make the message unpredictable and cumbersome for unit testing.
+	// We use []ints instead of []Reason to use `sort.Ints` without fuss.
+	var allowed []int
+	for reason, _ := range UserAllowedReasons {
+		allowed = append(allowed, int(reason))
+	}
+	sort.Ints(allowed)
+
+	// Find the string representation of each allowed reason code and build
+	// a message to return in `buf`.
+	var buf strings.Builder
```

Comment:
I think this could be simplified a bit, along the lines of
```
var stringified []string
for _, reason := range allowed {
	stringified = append(stringified, fmt.Sprintf("%s (%d)", ReasonToString[reason], reason))
}
return strings.Join(stringified, ", ")
```

Reply:
Good call, that's much nicer. Fixed in 5353688, thanks!